### PR TITLE
add wechat pay

### PR DESCRIPTION
**Pull request by simman**
add wechat pay

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -404,6 +404,28 @@ Receive result for `shareToTimeline` and `shareToSession` and arguments would be
...
+            package: '', // 商家根据财付通文档填写的数据和签名
+            sign: '' // 商家根据微信开放平台文档对数据做的签名
+          }
+      );
+      console.log('Pay for success!');
+    } catch (error) {
+      console.log('Pay for failure!');
+    }
+  }
+```
```
Comment (yorkie) [inline README.md]: Correct the indention to 2 spaces please, thank you

File: README.md [hunk 2]
```diff
@@ -404,6 +404,28 @@ Receive result for `shareToTimeline` and `shareToSession` and arguments would be
 
 For more details, visit [WeChat SDK].
 
+#### `pay`
+
+```
+async useWeChat() {
```
Comment (yorkie) [inline README.md]: Remove the wrapping function.

File: android/src/main/java/com/theweflex/react/WeChatModule.java [hunk 3]
```diff
@@ -441,6 +471,10 @@ public void onResp(BaseResp baseResp) {
         } else if (baseResp instanceof SendMessageToWX.Resp) {
             SendMessageToWX.Resp resp = (SendMessageToWX.Resp) (baseResp);
             map.putString("type", "SendMessageToWX.Resp");
+        } else if (baseResp instanceof PayResp) {
+            PayResp resp = (PayResp) (baseResp);
+            map.putString("type", "PayReq.Resp");
+            map.putString("returnkey", resp.returnKey);
```
Comment (yorkie) [inline android/src/main/java/com/theweflex/react/WeChatModule.java]: use "returnKey"

File: index.js [hunk 4]
```diff
@@ -183,3 +183,28 @@ export function shareToTimeline(data) {
...
+ * @param {String} data.prepayId
+ * @param {String} data.nonceStr
+ * @param {String} data.timeStamp
+ * @param {String} data.package
+ * @param {String} data.sign
+ * @returns {Promise}
+ */
+export function pay(data) {
+  return new Promise((resolve, reject) => {
+    WeChat.pay(data,() => {});
```
Comment (yorkie) [inline index.js]: This 2nd argument is not used correctly, propose:

```js
return new Promise((resolve, reject) => {
  WeChat.pay(data, () => resolve());
}).then(() => {
  emitter.on('PayReq.Resp', ...);
});
```

But I'm not sure if there is something problems when listening "PayReq.Resp" after the callback get invoked :)

File: android/src/main/java/com/theweflex/react/WeChatModule.java (hunk 3 above)
Comment (simman) [inline android/src/main/java/com/theweflex/react/WeChatModule.java]: What's the problem ?

Comment (yorkie) [inline android/src/main/java/com/theweflex/react/WeChatModule.java]: See [this line](https://github.com/simman/react-native-wechat/blob/213ba2010a2c12df944b04bc4a4f6dd16e2058bb/index.js#L204), it causes `resp.returnKey` to be undefined :)

Comment (simman) [inline android/src/main/java/com/theweflex/react/WeChatModule.java]: ok, I have corrected.

Comment (yorkie) [inline android/src/main/java/com/theweflex/react/WeChatModule.java]: Have corrected? Did you push your code in live? I don't see any changes on this PR :(

**Comment by simman** [comment]
@yorkie , I Have already submitted.

**Comment by yorkie** [comment]
All changes looks good to me, and I will keep this still open for 48 hours, and will merge and release if there are no any objection :)

**Pull request merged**